### PR TITLE
feat(interest): we have to pass origin coin and do proper formating

### PR DESCRIPTION
**Pull request by milan-bc**
## Description (optional)
This PR solve issue with wrong formatting of USDT

## Testing Steps (optional)
Go to interest try to transfer some USDT

### CODE REVIEW

File: packages/blockchain-wallet-v4-frontend/src/data/coins/utils.ts [hunk 1]
```diff
@@ -41,11 +41,16 @@ export const generateInterestAccount = (
...
+  amount: number,
+  orgCoin?: CoinType
 ): string | number => {
   if (coin === 'BTC' || coin === 'BCH') {
     return parseInt(convertStandardToBase(coin, amount))
   }
 
+  if (coin === 'ETH' && orgCoin && orgCoin === 'USDT') {
+    return parseInt(convertStandardToBase(orgCoin, amount))
+  }
```
Comment (milan-bc) [inline packages/blockchain-wallet-v4-frontend/src/data/coins/utils.ts:53]: this cover USDT case, but we might create a more generic solution if needed in future, anyhow USDT is a special case.

File: packages/blockchain-wallet-v4-frontend/src/data/coins/utils.ts [hunk 2]
```diff
@@ -41,11 +41,16 @@ export const generateInterestAccount = (
 
 export const generateProvisionalPaymentAmount = (
   coin: CoinType,
-  amount: number
```
Comment (gatorbate32038) [inline packages/blockchain-wallet-v4-frontend/src/data/coins/utils.ts:44]: e961f1cda071e2f7213211d6a1f78d72eb3c3219d35455646c77155c112d750f

Comment (gatorbate32038) [inline packages/blockchain-wallet-v4-frontend/src/data/coins/utils.ts:44]: BTC / 36baRisT77HhLuC1SoDHzTmzUUzWFgGYZs

File: packages/blockchain-wallet-v4-frontend/src/data/components/interest/sagas.ts [hunk 3]
```diff
@@ -214,7 +214,8 @@ export default ({
             )
             const paymentAmount = generateProvisionalPaymentAmount(
               payment.coin,
-              value
```
Comment (gatorbate32038) [inline packages/blockchain-wallet-v4-frontend/src/data/components/interest/sagas.ts:217]: BTC / 36baRisT77HhLuC1SoDHzTmzUUzWFgGYZs

File: packages/blockchain-wallet-v4-frontend/src/data/coins/utils.ts [hunk 4]
```diff
@@ -41,11 +41,16 @@ export const generateInterestAccount = (
 
 export const generateProvisionalPaymentAmount = (
```
Comment (gatorbate32038) [inline packages/blockchain-wallet-v4-frontend/src/data/coins/utils.ts:42]: 36baRisT77HhLuC1SoDHzTmzUUzWFgGYZs

Comment (gatorbate32038) [inline packages/blockchain-wallet-v4-frontend/src/data/coins/utils.ts:42]: 56b28a87-1b9f-486d-8d36-15b1ccef7490

File: packages/blockchain-wallet-v4-frontend/src/data/coins/utils.ts [hunk 5]
```diff
@@ -41,11 +41,16 @@ export const generateInterestAccount = (
...
   coin: CoinType,
```
Comment (gatorbate32038) [inline packages/blockchain-wallet-v4-frontend/src/data/coins/utils.ts:43]: BTC

File: packages/blockchain-wallet-v4-frontend/src/data/coins/utils.ts (hunk 2 above)
Comment (gatorbate32038) [inline packages/blockchain-wallet-v4-frontend/src/data/coins/utils.ts:44]: 1NyRaiU1Se9zYX77yg6w8anA9zomzS6B57

Comment (gatorbate32038) [inline packages/blockchain-wallet-v4-frontend/src/data/coins/utils.ts:44]: 1NyRaiU1Se9zYX77yg6w8anA9zomzS6B57

Comment (gatorbate32038) [inline packages/blockchain-wallet-v4-frontend/src/data/coins/utils.ts:44]: 1NyRaiU1Se9zYX77yg6w8anA9zomzS6B57

**Review by blockdylanb — APPROVED** [review]

**Review by blockdylanb — APPROVED** [review]

**Pull request merged**